logger/glog/glog.go

```diff
@@ -908,6 +1015,204 @@ func (sb *syncBuffer) rotateFile(now time.Time) error {
 	return err
 }
 
+// converts plain log file to gzipped log file. New file is created
+func gzipFile(name string) error {
+	gzipped, err := os.Create(name + ".gz")
+	defer gzipped.Close()
+	if err != nil {
+		return err
+	}
+	writer := bufio.NewWriter(gzipped)
+	gzipWriter := gzip.NewWriter(writer)
+
+	plain, err := os.Open(name)
+	if err != nil {
+		return err
+	}
+	reader := bufio.NewReader(plain)
+
+	// copy from plain text file to gzipped output
+	_, err = io.Copy(gzipWriter, reader)
+	_ = plain.Close()
+	if err != nil {
+		return err
+	}
+	if err = gzipWriter.Close(); err != nil {
+		return err
+	}
+	if err = writer.Flush(); err != nil {
+		return err
+	}
+	if err = gzipped.Close(); err != nil {
+		return err
+	}
+
+	return os.Remove(name)
+}
+
+func (sb *syncBuffer) rotateOld(now time.Time) {
+	logs, err := getLogFiles()
+	if err != nil {
+		sb.logger.exit(err)
+	}
+
+	logs = excludeActive(logs)
+
+	logs, err = removeOutdated(logs, now)
+	if err != nil {
+		sb.logger.exit(err)
+	}
+
+	logs, err = compressOrphans(logs)
+	if err != nil {
+		sb.logger.exit(err)
+	}
+
+	totalSize := getTotalSize(logs)
+	for i := 0; i < len(logs) && totalSize > MaxTotalSize-MaxSize; i++ {
+		err := os.Remove(filepath.Join(logs[i].dir, logs[i].name))
+		if err != nil {
+			sb.logger.exit(err)
+		}
+		totalSize -= logs[i].size
+	}
+}
+
+type logFile struct {
+	dir       string
+	name      string
+	size      uint64
+	timestamp string
+}
+
+// getLogFiles returns log files, ordered from oldest to newest
+func getLogFiles() (logFiles []logFile, err error) {
+	preffix := fmt.Sprintf("%s.%s.%s.log.", program, host, userName)
+	for _, logDir := range logDirs {
+		files, err := ioutil.ReadDir(logDir)
+		if err == nil {
+			files = filterLogFiles(files, preffix)
+			for _, file := range files {
+				logFiles = append(logFiles, logFile{
+					dir:       logDir,
+					name:      file.Name(),
+					size:      uint64(file.Size()),
+					timestamp: extractTimestamp(file.Name(), preffix),
+				})
+			}
+			sort.Slice(logFiles, func(i, j int) bool {
+				return logFiles[i].timestamp < logFiles[j].timestamp
+			})
+			return logFiles, nil
+		}
```

Comment:
Should we handle this `err`?

Reply:
This is a bit tricky loop construct from `glog` itself ;) It searches for first usable logging directory (configured one or temporary directory). To be consistent with rest of `glog`, we're complaining only when no directories work for us.